### ISSUE TITLE
Click event does not fire on certain mobile browsers

### ISSUE THREAD

**Issue by soberdor** [openlayers/openlayers#14155]
When handling feature selection, on a few browsers (confirmed eg on two different iPhone 8 with iOS 16, but reportedly also some Android devices) **the click event does not fire when touching the map**. 

Normal sequence of events: pointerdown -> pointerup -> ***click*** -> singleclick
On these devices: pointerdown -> ***pointermove*** -> pointerup -> singleclick

Of note, on these iPhones _disabling the 3D touch feature_ (meanwhile discontinued by Apple on newer iPhones) solved the issue.

I tried to gradually increase the moveTolerance for the map up to extreme values, but this did _not_ resolve the issue (while having an obvious impact on map panning).

![grafik](https://user-images.githubusercontent.com/27883798/193408515-63444c25-8505-4eda-9617-4447e1e23ff7.png)

**Comment by ahocevar** [comment]
Are you experiencing this also with the latest version (v7.1.0) of OpenLayers?

**Comment by soberdor** [comment]
Yes, it’s the same behaviour. When it was reported to me a couple of weeks ago, I was running OL v6.3.1, but no changes after upgrading to 7.1.0.

(for now I helped myself with a fallback to singleClick if there was no click event)

**Comment by ahocevar** [comment]
That's interesting, because there is no code path that fires a singleclick event without a click.

**Comment by ahocevar** [comment]
The problem can indeed be reproduced in a test. Working on a fix now. See #14157.

**Comment by ahocevar** [comment]
Update: Actually the test I wrote in #14157 was wrong, because I dispatched the move and up events on the element instead of the document.

@soberdor Does this occur with any of the official examples, e.g. https://openlayers.org/en/latest/examples/icon.html? Could it be that this issue occurs when a shadow DOM is used for the map? What happens when you click on the red circle with the devices in question on https://codesandbox.io/s/es2015-custom-element-forked-ugpbsq?file=/main.js ?

**Comment by soberdor** [comment]
Thank you for looking into this. I finally got to have these examples tested on the affected devices - **and they work!**

However, these examples both use a **click event attached to the map object**, as in
`
    this.map.on("click", (e) => {...});
`

Whereas the described behaviour occurs **within the condition function of a select interaction**:

```
const myAnnotationSelector = new ol.interaction.Select({
  hitTolerance: 15,
    condition: function (e) {
      console.log(e.type);
    },
    layers: [annotationLayer]
});
```

**Comment by ahocevar** [comment]
Ok, I had missed the part "When handling feature selection within the condition function of a select interaction" in the original issue description.

Well, as the name of the `condition` option suggests, it is function designed to return a boolean value to indicate whether to perform an action or not. It is **not** meant to have side effects. So I'd say the reported problem is not a library bug, but a problem resulting from performing side effects in a condition function.

**Comment by soberdor** [comment]
This is actually what I use the `condition` for: to return a boolean whether a select event should be fired on the interaction - is this the intended use? I understood it like this. 

This is the condition's return value:
`return ol.events.condition.click(e);`
But this statement will never amount to `true` on the devices mentioned above.

So shouldn't there also be an `e.type='click'` coming along? This is what happens on virtually all devices except the ones mentioned. Is this intended? At the very least it doesn't seem consistent.

**Comment by ahocevar** [comment]
Ok, I see now what you are trying to do. Yes, the intense use is to decide whether selection should occur or not.

Could it be that the application in question has another interaction that handles click events and stops the event chain?

**Comment by ahocevar** [comment]
Can you reproduce the problem on https://openlayers.org/en/latest/examples/select-features.html ?

**Comment by soberdor** [comment]
I had the example tested, and the behaviour could not be reproduced. I created a modified code sandbox, which shows the same differences in the sequence of events (pointerdown -> _pointermove_ -> pointerup), but most importantly in spite of the _pointermove_, the _click_ event follows and the interaction's _select_ event is fired.

https://cwql1x.csb.app/

![grafik](https://user-images.githubusercontent.com/27883798/196244498-49f67499-6074-4745-a78c-c674f0d04a81.png)

I will have to look deeper into my app, but I'm not aware of any explicit stopPropagation calls (would another interaction stop the event chain by default?). Also, I still can't think of any reason why the problem only appears on very few devices.
